### ISSUE TITLE
[rllib] Installing python-snappy doesn't work reliably.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#1781]
RLlib lists `python-snappy` as a requirement

https://github.com/ray-project/ray/blob/7c4afa4b046c988f398ab5132f033cd78619bca4/python/setup.py#L55-L57

However, when I do `pip install python-snappy` I always get errors like the following.

What is the recommended way to install `python-snappy`? Do we really want to make this a requirement for RLlib?

```
$ pip install python-snappy
Collecting python-snappy
  Using cached python-snappy-0.5.2.tar.gz
Building wheels for collected packages: python-snappy
  Running setup.py bdist_wheel for python-snappy ... error
  Complete output from command /home/ubuntu/anaconda3/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-piqy6lxi/python-snappy/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" bdist_wheel -d /tmp/tmp8rz0qg0upip-wheel- --python-tag cp35:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.5
  creating build/lib.linux-x86_64-3.5/snappy
  copying snappy/__main__.py -> build/lib.linux-x86_64-3.5/snappy
  copying snappy/snappy.py -> build/lib.linux-x86_64-3.5/snappy
  copying snappy/snappy_cffi_builder.py -> build/lib.linux-x86_64-3.5/snappy
  copying snappy/snappy_cffi.py -> build/lib.linux-x86_64-3.5/snappy
  copying snappy/hadoop_snappy.py -> build/lib.linux-x86_64-3.5/snappy
  copying snappy/snappy_formats.py -> build/lib.linux-x86_64-3.5/snappy
  copying snappy/__init__.py -> build/lib.linux-x86_64-3.5/snappy
  running build_ext
  building 'snappy._snappy' extension
  creating build/temp.linux-x86_64-3.5
  creating build/temp.linux-x86_64-3.5/snappy
  gcc -pthread -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/ubuntu/anaconda3/include/python3.5m -c snappy/snappymodule.cc -o build/temp.linux-x86_64-3.5/snappy/snappymodule.o
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  snappy/snappymodule.cc:31:22: fatal error: snappy-c.h: No such file or directory
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for python-snappy
  Running setup.py clean for python-snappy
Failed to build python-snappy
Installing collected packages: python-snappy
  Running setup.py install for python-snappy ... error
    Complete output from command /home/ubuntu/anaconda3/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-piqy6lxi/python-snappy/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-egrr_2a2-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.5
    creating build/lib.linux-x86_64-3.5/snappy
    copying snappy/__main__.py -> build/lib.linux-x86_64-3.5/snappy
    copying snappy/snappy.py -> build/lib.linux-x86_64-3.5/snappy
    copying snappy/snappy_cffi_builder.py -> build/lib.linux-x86_64-3.5/snappy
    copying snappy/snappy_cffi.py -> build/lib.linux-x86_64-3.5/snappy
    copying snappy/hadoop_snappy.py -> build/lib.linux-x86_64-3.5/snappy
    copying snappy/snappy_formats.py -> build/lib.linux-x86_64-3.5/snappy
    copying snappy/__init__.py -> build/lib.linux-x86_64-3.5/snappy
    running build_ext
    building 'snappy._snappy' extension
    creating build/temp.linux-x86_64-3.5
    creating build/temp.linux-x86_64-3.5/snappy
    gcc -pthread -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/ubuntu/anaconda3/include/python3.5m -c snappy/snappymodule.cc -o build/temp.linux-x86_64-3.5/snappy/snappymodule.o
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    snappy/snappymodule.cc:31:22: fatal error: snappy-c.h: No such file or directory
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/home/ubuntu/anaconda3/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-piqy6lxi/python-snappy/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-egrr_2a2-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-piqy6lxi/python-snappy/
```

@ericl @richardliaw

**Comment by ericl** [comment]
We could make it optional, or look for an alternative such as lz4. A fast compressor is important for replay buffer performance.

**Comment by ericl** [comment]
Note that snappy is already a soft dependency so it doesn't need to be installed for rllib to run.

**Comment by richardliaw** [comment]
Does the `rllib` installation fail if snappy is not installed correctly? If so, we should remove it from the setup script, right?

**Comment by robertnishihara** [comment]
Yes, that's right, it should not be in the `setup.py` because it makes ` pip install ray[rllib]` not work.

**Comment by robertnishihara** [comment]
Fixed by #1847.
